src/main/java/seedu/address/commons/core/FunctionalBrowser.java

```diff
@@ -51,17 +71,31 @@ public static FunctionalBrowser getDesktop() {
      *      or the default handler application failed to be launched
      * @throws CommandException Generated URI is invalid.
      */
-    public void launchUri(String url) throws IOException, CommandException {
+    public void launchUri(String url) throws CommandException {
         requireNonNull(url);
         try {
             URI uri = new URI(url);
-            if (Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)) {
-                this.desktop.browse(uri);
+            if (isLinux()) {
+                //For linux OS
+                if (Runtime.getRuntime().exec(new String[] { "which", "xdg-open" }).getInputStream().read() != -1) {
+                    Runtime.getRuntime().exec(new String[] { "xdg-open", url });
+                } else {
+                    throw new CommandException("The current OS does not support opening external links.");
+                }
             } else {
-                throw new CommandException("The current OS does not support browser extensions");
+                //For non linux OS
+                if (Desktop.isDesktopSupported() && Desktop.getDesktop().isSupported(Desktop.Action.BROWSE)) {
+                    this.desktop.browse(uri);
```

Comment:
Not sure if there is a chance where `desktop` is null. Maybe a null check would be better but I think its fine without as well.

Reply:
I think let's stay on the safe side and include it 🤣